statistics/handle/cache/internal/lfu/lfu_cache.go

```diff
@@ -130,3 +148,9 @@ func (s *LFU) wait() {
 func (s *LFU) metrics() *ristretto.Metrics {
 	return s.cache.Metrics
 }
+
+// Stop implements statsCacheInner
+func (s *LFU) Stop() {
+	s.cache.Close()
+	s.cache.Wait()
```

Comment:
Would this possibly wait for a long time?

Reply:
No, the goroutine in the cache will receive the close signal and then exit.

Reply:
Will it cause some dead-lock or dead-goroutine if we call `cache.Close()` before `cache.Wait()`?

Reply:
Let's start by looking at the implementation of ```wait``` and ```close```. 

- ```Close``` is at the https://github.com/dgraph-io/ristretto/blob/main/cache.go#L354.

Close is to close some channels to send a close signal. so that the background task will be stopped like evict policy. Finally, it will be marked as closed.

- ```Wait``` is at the https://github.com/dgraph-io/ristretto/blob/main/cache.go#L213

 It seems that "wait" doesn't have any effect because once you close it, it returns immediately.

Reply:
it has been marked as closed. All the ```get```/```set``` will return nothing immediately.